src/main/java/tdl/s3/upload/MultipartUploadFileUploadingStrategy.java

```diff
@@ -128,13 +131,18 @@ private void uploadRequiredParts(File file, String remotePath) throws IOExceptio
     private Stream<UploadPartRequest> streamUploadForFailedParts(File file, String remotePath) {
         return failedMiddleParts.stream()
                 .map(partNumber -> {
-                    byte[] partData = ByteHelper.readPart(partNumber, file);
-                    uploadedSize += partData.length;
-                    return getUploadPartRequest(remotePath, partData, false, partNumber);
-                });
-    }
-
-    private Stream<UploadPartRequest> streamUploadForIncompleteParts(String remotePath, InputStream inputStream, boolean writingFinished) throws IOException {
+                    try {
+                        byte[] partData = ByteHelper.readPart(partNumber, file);
+                        UploadPartRequest request = getUploadPartRequest(remotePath, partData, false, partNumber);
+                        uploadedSize += partData.length;
+                        return request;
+                    } catch (IOException | DestinationOperationException ex) {
+                        return null;
```

Comment:
Yes, now that you mentioned it. I think that maintaining a state `uploadedSize` in a process like this kind of defeat the purpose of functional programming.